### PR TITLE
NDBE and ZTP integration

### PR DESCRIPTION
**Pull request by lack**
- ztp: SiteConfig spec and example changes for Disk Encryption
- ztp: Generate optional disk encryption manifest

Before merge:
- [x] Finalize schema design
- [x] Demonstrate empty and disabled config do no harm
- [x] Happy path install test
- [x] Failed install test

### CODE REVIEW

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum [hunk 1]
```diff
@@ -1,4 +1,12 @@
+github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
```
[the remark this answers is not in the document]
Comment (lack) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum:1]: These are all dependencies of `github.com/stretches/testify`, which is a great and widely used MIT-licensed convenience library for test assertions.  Without it, unit tests have a lot more boilerplate.

There's no go mod way to differentiate, but these are only used by the unit tests and not the actual PolicyGenerator itself.

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
+github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
+github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
```
[the remark this answers is not in the document]
Comment (lack) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum:5]: It looks like MIT to me: https://github.com/stretchr/objx/blob/master/LICENSE

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum (hunk 1 above)
[the remark this answers is not in the document]
Comment (nishant-parekh) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/go.sum:1]: I am using testify for the tests I added as well.

**Comment by lack** [comment]
I have verified that leaving out the new `diskEncryption` section, or adding it with `type: "none"` produces identical site config customResources as the current head of the master branch.

**Comment by lack** [comment]
I've also verified that the siteConfig-generated customResources can successfully deploy an SNO.

**Comment by lack** [comment]
With the additional systemd unit in the NBDE manifest, this installs NBDE successfully on my lab system.

**Comment by lack** [comment]
When the NBDE information is wrong or the tang servers are offline, installation fails.  Basically the machine keeps rebooting as it continually tries to get ignition to complete the disk encryption phase, and eventually Assisted Service times out and marks it as "error".

**Comment by lack** [comment]
Rebased to take into account @serngawy 's latest update to allow custom manifests.

**Review by serngawy — APPROVED** [review]
Thanks Jim, I'm approving this PR and later we can discuss more how to make the go tmpl more generic to fit all cases

**Pull request merged**